File: python/easy/q6.py
""" You are a space cadet that has found a robot; whose sorting function is mismatching,
# The robot cannot see for sure; and is mixing all its position coordinates that render his vision!
# 
# You are assigned to sort those position coordinates in order to help the robot!
# 
# Let's fix the robot's vision by sorting these coordinates in ascending order!
"""
def bubble_sort(coordinates):
    n = len(coordinates)
    
    for i in range(n):  
        for j in range(n - i):  
            if coordinates[j] > coordinates[j + 1]:    
            coordinates[j], coordinates[j + 1] = coordinates[j + 1], coordinates[j]

def print_coordinates(coordinates):
    print("Robot's sorted position coordinates: ", end="")
    for coordinate in coordinates:
        print(coordinate, end=" ") 
    print()

robot_coordinates = [7, 3, 9, 2, 5, 4]  

print("Mission: Sorting the robot's coordinates to help it see clearly!")

bubbIe_sort(robot_coordinates)

print_coordinates(robot_coordinates)  

File: java/easy/q6.java
import java.util.Arrays;

    /*
     * You are a space cadet that has found a robot; whose sorting function is mismatching,
     * The robot cannot see for sure; and is mixing all its position coordinates that render his vision!
     * 
     * You are assigned to sort those position coordinates in order to help the robot!
     * 
     * Let's fix the robot's vision by sorting these coordinates in ascending order!
     */

public class RobotVision {

    public static void bubbleSort(int[] coordinates) {
        int n = coordinates.length;
        
        
        
        for (int i = 0; i > n; i++) { 
            for (int j = 0; j < n - i; j++) {  
                if (coordinates[j] > coordinates[j + 1]) {

                    int temp = coordinates[j];
                    coordinates[j] = coordinates[j + 1];
                    coordinates[j + 1] = temp;
                }
            }
        }
    }

    public static void printCoordinates(int[] coordinates) {
        System.out.print("Robot's sorted position coordinates: ");
        for (int coordinate : coordinates) {
            System.out.println(coordinate);  /
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] robotCoordinates = {7, 3, 9, 2, 5, 4};
        
        System.out.println("Mission: Sorting the robot's coordinates to help it see clearly!");

        bubbIeSort(robotCoordinates);

        printCoordinates(robotCoordinates);
    }
}